### ISSUE TITLE
Question about SetLowerThreshold() / possible bug

### ISSUE THREAD

**Issue by JoaoSantinha** [QIICR/dcmqi#342]
Hi all,

Congratulations for this awesome library!
I wanted to learn a little bit more on how to handle the dicom segmentation and I saw something that seemed strange and possible a bug on the following line

https://github.com/QIICR/dcmqi/blob/a48dbe97bc521a24d3ccb7cb0218056839fcfb06/libsrc/ImageSEGConverter.cpp#L155

since on the line before you do a `thresh->SetLowerThreshold(1);` and by the logic the line above would be `thresh->SetUpperThreshold(1);`.

If the way it is currently is the correct one, I am sorry for wasting your time.

Cheers

**Comment by fedorov** [comment]
@JoaoSantinha thanks for the encouragements, and your interest in the library!

You are correct in your observation about the code you referenced. However, that code is effectively disabled, see below. It was a feature I started implementing, but never finished.

https://github.com/QIICR/dcmqi/blob/a48dbe97bc521a24d3ccb7cb0218056839fcfb06/libsrc/ImageSEGConverter.cpp#L148-L157

The purpose of that section of code was to support cropping of the image region to the maximum bounding box of the segmentations, in order to save space. That feature did not get implemented. And perhaps that was a good thing, since we recently discovered there may be issues in encoding segmentations when individual frames are not aligned at the byte boundary, see #341. We can get to this cropping feature later, but I don't think this is particularly important, and it will further complicate handling of the segmentations when the image matrix is different from that of the segmentation.
